File: Assignment32_5.c
#include <stdio.h>

//////////////////////////////////////////////////////////////
// Function Name : Pattern
// Description   : Display diagonal pattern
// Input         : Integer (Rows, Columns)
// Output        : Pattern display
// Author        : Raj Samir Jadhav
// Date          : 26/11/2025
//////////////////////////////////////////////////////////////

void Pattern(int iRow, int iCol)
{
    int i = 0, j = 0;

    for(i = 1; i <= iRow; i++)
    {
        for(j = 1; j <= iCol + 1; j++)
        {
            if(i == 1 || i == iRow)          // First or last row
            {
                printf("%d\t", j);
            }
            else if(j == 1)                  // First column
            {
                printf("1\t");
            }
            else if(j == 2)                  // Second column
            {
                printf("2\t");
            }
            else if(j == iCol + 1)           // Last column 
            {
                printf("5\t");
            }
            else if(i == j)                  // Diagonal
            {
                printf("%d\t", i);
            }
            else
            {
                printf("\t");                // Blank
            }
        }
        printf("\n");
    }
}

int main()
{
    int iValue1 = 0, iValue2 = 0;

    printf("Enter number of rows:\n");
    scanf("%d", &iValue1);

    printf("Enter number of columns:\n");
    scanf("%d", &iValue2);

    Pattern(iValue1, iValue2);
    return 0;
}

File: Assignment32_5.java
import java.util.Scanner;

//////////////////////////////////////////////////////////////
// Function Name : Pattern
// Description   : Display  diagonal pattern
// Input         : Integer (Rows, Columns)
// Output        : Pattern display
// Author        : Raj Samir Jadhav
// Date          : 26/11/2025
//////////////////////////////////////////////////////////////

class Pattern
{
    public void Display(int iRow, int iCol)
    {
        for(int i = 1; i <= iRow; i++)
        {
            for(int j = 1; j <= iCol + 1; j++)
            {
                if(i == 1 || i == iRow)        // First or last row
                {
                    System.out.print(j + "\t");
                }
                else if(j == 1)                // First column
                {
                    System.out.print("1\t");
                }
                else if(j == 2)                // Second column
                {
                    System.out.print("2\t");
                }
                else if(j == iCol + 1)         // Last column
                {
                    System.out.print("5\t");
                }
                else if(i == j)                // Diagonal
                {
                    System.out.print(i + "\t");
                }
                else
                {
                    System.out.print("\t");     // Blank
                }
            }
            System.out.println();
        }
    }
}

class Assignment32_5
{
    public static void main(String[] args)
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter number of rows:");
        int iValue1 = sobj.nextInt();

        System.out.println("Enter number of columns:");
        int iValue2 = sobj.nextInt();

        Pattern pobj = new Pattern();
        pobj.Display(iValue1, iValue2);
    }
}
